core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -321,13 +327,20 @@ touch (ArrayContents contents) =
 
 fptrToArrayContents :: ForeignPtrContents -> ArrayContents
 arrayToFptrContents :: ArrayContents -> ForeignPtrContents
+arrContentsToMutByteArr :: ArrayContents -> MutableByteArray# RealWorld
+byteArrToArrContents :: MutableByteArray# RealWorld -> ArrayContents
 #ifdef USE_FOREIGN_PTR
 fptrToArrayContents = ArrayContents
 arrayToFptrContents (ArrayContents contents) = contents
+arrContentsToMutByteArr (ArrayContents (PlainPtr mbarr)) = mbarr
+arrContentsToMutByteArr (ArrayContents _) = error "Unsupported foreign ptr"
+byteArrToArrContents mbarr = (ArrayContents (PlainPtr mbarr))
```

Comment:
In case of foreign ptr we can make pin/unpin as no-op instead of erroring out.

Reply:
Will that not be a little deciving?